test/integration/rtds_integration_test.cc

```diff
@@ -140,6 +144,70 @@ class RtdsIntegrationTest : public Grpc::DeltaSotwIntegrationParamTest, public H
 INSTANTIATE_TEST_SUITE_P(IpVersionsClientTypeDelta, RtdsIntegrationTest,
                          DELTA_SOTW_GRPC_CLIENT_INTEGRATION_PARAMS);
 
+#ifndef WIN32
+// TODO): The directory rotation via TestEnvironment::renameFile() fails on Windows. The
+// renameFile() implementation does not correctly handle symlinks.
+
+// This test mimics what K8s does when it swaps a ConfigMap.
+TEST_P(RtdsIntegrationTest, FileRtdsReload) {
+  // Create an initial setup that looks similar to a K8s ConfigMap deployment. This is a file
+  // contained in a directory and referenced via an intermediate symlink on the directory.
+  const std::string temp_path{TestEnvironment::temporaryDirectory() + "/rtds_test"};
+  TestEnvironment::createPath(temp_path + "/data_1");
+  TestEnvironment::writeStringToFileForTest(temp_path + "/data_1/rtds.yaml", R"EOF(
+resources:
+- "@type": type.googleapis.com/envoy.service.runtime.v3.Runtime
+  name: file_rtds
+  layer:
+ )EOF",
+                                            true);
+  TestEnvironment::createSymlink(temp_path + "/data_1", temp_path + "/..data");
+  TestEnvironment::createSymlink(temp_path + "/..data/rtds.yaml", temp_path + "/rtds.yaml");
```

Comment:
Maybe some comment with the resulting directory structure would make it easier to follow what this ends up creating?